collated/main/A0161247J.md

```diff
@@ -0,0 +1,637 @@
+				toReturn = createDeadlineTask(args);
+			} else if (isFloatingCommand(args)) {
+				toReturn = createFloatingTask(args);
+			} else {
+				throw new IllegalArgumentException();
+			}
+		} catch (NullPointerException e) {
+        	hasException = true;
+        } catch (ParseException e) {
+        	hasException = true;
+        } catch (IndexOutOfBoundsException e) {
+        	hasException = true;
+        } catch (IllegalArgumentException e) {
+        	hasException = true;
+        } catch (IllegalValueException e) {
+            hasException = true;
+        } catch (DateTimeParseException e) {
+            hasException = true;
+        }
+		
+		if (hasException) {
+        	toReturn = new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, 
+                    		EditCommand.MESSAGE_USAGE));
+        }
+		
+		return toReturn;
+    }
+	
+	/**
+	 * Method used to retrieve the index from a string argument 
+	 * @param argIndex
+	 * @throws NullPointerException
+	 * @throws IllegalArgumentException
+	 */
+	private int tryParseIndex(String argIndex) throws NullPointerException, IllegalArgumentException {
+        Optional<Integer> index = parseIndex(argIndex);
+        if(!index.isPresent()){
+            throw new IllegalArgumentException();
+        }
+        
+        return index.get();
+	}
+	
+	private boolean isDeadlineCommand(String args) throws NullPointerException {
+		final Matcher matcher = DEADLINE_ARGS_FORMAT.matcher(args);
+		return matcher.matches();
+	}
+	
+	private boolean isEventCommand(String args) throws NullPointerException {
+		final Matcher matcher = EVENT_ARGS_FORMAT.matcher(args);
+		return matcher.matches();
+	}
+	
+	private boolean isFloatingCommand(String args) throws NullPointerException {
+		final Matcher matcher = FLOATING_ARGS_FORMAT.matcher(args);
+		return matcher.matches();
+	}
+	
+	/**
+	 * Creates a EditCommand for a DeadlineTask given a list of arguments expects args to have the form
+	 * "[INDEX] 00-00-0000"
+	 * 
+	 * @throws ParseException 
+	 * @throws IllegalArgumentException
+	 */
+	private Command createDeadlineTask(String args) throws IllegalArgumentException, ParseException, DateTimeParseException {
+		Matcher matcher = DEADLINE_ARGS_FORMAT.matcher(args);
+		
+		if (!matcher.matches()) {
+			throw new IllegalArgumentException();
+		}
+		
+		String indexString = matcher.group("index");
+		String endDateString = matcher.group("endDate");
+		String endTimeString = matcher.group("endTime");
+		
+		LocalDateTime endDate = (endTimeString == null) ? 
+			DateUtil.parseStringToLocalDate(endDateString) :
+			DateUtil.parseStringToLocalDateTime(endDateString + " " + endTimeString);
+		
+		int index = tryParseIndex(indexString);
+        return new EditCommand(index, new TaskDate(endDate));
+	}
+	
+	/**
+	 * Creates an EditCommand for an EventTask given a string argument that has the form
+	 * "[INDEX] 00-00-0000 00-00-0000"
+	 * 
+	 * @throws ParseException
```

Comment:
please give a brief description of why? in its current form the `@throws` line is not providing any additional information